// smart-contracts-boilerplate/typechained/factories/KhooleeCoinMinterInterface__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type {
  KhooleeCoinMinterInterface,
  KhooleeCoinMinterInterfaceInterface,
} from "../KhooleeCoinMinterInterface";

const _abi = [
  {
    inputs: [
      {
        internalType: "address",
        name: "owner",
        type: "address",
      },
    ],
    name: "balanceOf",
    outputs: [
      {
        internalType: "uint256",
        name: "",
        type: "uint256",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "account",
        type: "address",
      },
      {
        internalType: "uint256",
        name: "amount",
        type: "uint256",
      },
    ],
    name: "burnFrom",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "to",
        type: "address",
      },
      {
        internalType: "uint256",
        name: "amount",
        type: "uint256",
      },
    ],
    name: "mint",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
];

export class KhooleeCoinMinterInterface__factory {
  static readonly abi = _abi;
  static createInterface(): KhooleeCoinMinterInterfaceInterface {
    return new utils.Interface(_abi) as KhooleeCoinMinterInterfaceInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): KhooleeCoinMinterInterface {
    return new Contract(
      address,
      _abi,
      signerOrProvider
    ) as KhooleeCoinMinterInterface;
  }
}
