// smart-contracts-boilerplate/typechained/factories/VaultContractInterface__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type {
  VaultContractInterface,
  VaultContractInterfaceInterface,
} from "../VaultContractInterface";

const _abi = [
  {
    inputs: [
      {
        internalType: "uint256",
        name: "collateralAmount",
        type: "uint256",
      },
    ],
    name: "addCollateral",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "uint256",
        name: "borrowAmount",
        type: "uint256",
      },
    ],
    name: "borrowKhooleeCoins",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "uint256",
        name: "collateralAmount",
        type: "uint256",
      },
    ],
    name: "removeCollateral",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
];

export class VaultContractInterface__factory {
  static readonly abi = _abi;
  static createInterface(): VaultContractInterfaceInterface {
    return new utils.Interface(_abi) as VaultContractInterfaceInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): VaultContractInterface {
    return new Contract(
      address,
      _abi,
      signerOrProvider
    ) as VaultContractInterface;
  }
}
